testsuite/MDAnalysisTests/analysis/test_msd.py

```diff
@@ -0,0 +1,218 @@
+        m_simple = MSD(u, SELECTION, msd_type=dim, fft=False)
+        m_simple.run()
+        timeseries_simple = m_simple.timeseries
+        m_fft = MSD(u,SELECTION, msd_type=dim, fft=True)
+        m_fft.run()
+        timeseries_fft = m_fft.timeseries
+        assert_almost_equal(timeseries_simple, timeseries_fft, decimal=4)
+
+def test_fft_vs_simple_all_dims_per_particle(dimension_list, u):
+    for dim in dimension_list:
+        m_simple = MSD(u, SELECTION, msd_type=dim, fft=False)
+        m_simple.run()
+        per_particle_simple = m_simple.msd_per_particle
+        m_fft = MSD(u,SELECTION, msd_type=dim, fft=True)
+        m_fft.run()
+        per_particle_fft = m_fft.msd_per_particle
+        assert_almost_equal(per_particle_simple, per_particle_fft, decimal=4)
+
+#testing on step trajectory in tidynamics
+def test_simple_step_traj_3d(step_traj): # this should fit the polynomial 3x**2
+    m_simple = MSD(step_traj, 'all' , msd_type='xyz', fft=False)
+    m_simple.run()
+    poly3 = characteristic_poly(NSTEP,3)
+    assert_almost_equal(m_simple.timeseries, poly3, decimal=4)
+
+def test_simple_step_traj_2d(step_traj): # this should fit the polynomial 2x**2
+    m_simple = MSD(step_traj, 'all' , msd_type='xy',  fft=False)
+    m_simple.run()
+    poly2 = characteristic_poly(NSTEP,2)
+    assert_almost_equal(m_simple.timeseries, poly2, decimal=4)
+
+def test_simple_step_traj_1d(step_traj): # this should fit the polynomial x**
+    m_simple = MSD(step_traj, 'all' , msd_type='x', fft=False)
+    m_simple.run()
+    poly1 = characteristic_poly(NSTEP,1)
+    assert_almost_equal(m_simple.timeseries, poly1,decimal=4)  
+
+def test_fft_step_traj_3d(step_traj): # this should fit the polynomial 3x**2
+    m_fft = MSD(step_traj, 'all' , msd_type='xyz', fft=True)
+    m_fft.run()
+    poly3 = characteristic_poly(NSTEP,3)
+    assert_almost_equal(m_fft.timeseries, poly3, decimal=3) # this was relaxed from decimal=4 for numpy=1.13 test
+
+def test_fft_step_traj_2d(step_traj): # this should fit the polynomial 2x**2
+    m_fft = MSD(step_traj, 'all' , msd_type='xy', fft=True)
+    m_fft.run()
+    poly2 = characteristic_poly(NSTEP,2)
+    assert_almost_equal(m_fft.timeseries, poly2, decimal=3) # this was relaxed from decimal=4 for numpy=1.13 test
+
+def test_fft_step_traj_1d(step_traj): # this should fit the polynomial x**2
+    m_fft = MSD(step_traj, 'all' , msd_type='x', fft=True)
+    m_fft.run()
+    poly1 = characteristic_poly(NSTEP,1)
+    assert_almost_equal(m_fft.timeseries, poly1, decimal=3) # this was relaxed from decimal=4 for numpy=1.13 test
+
+# test tidynamics on a constant velocity trajectory.
+def test_step_traj_tidy_3d(step_traj_arr): # this should fit the polynomial y = 3x**2
+    msd_tidy = tidynamics.msd(step_traj_arr.astype(np.float64))
+    print(msd_tidy)
+    poly3 = characteristic_poly(NSTEP,3)
+    print(poly3)
+    assert_almost_equal(msd_tidy, poly3, decimal=4) # this passes with NSTEP=5000, and 10000 but fails for 20,000
+
```

Comment:
@pdebuyl I have added this test to show some inaccuracies at high sample sizes. It is based on a constant velocity 3D trajectory whose MSD should fit y=3x**2.

This test fails on my architecture but may not on yours. Try playing with `NSTEP` at the top of the file (number of frames in the toy trajectory) and `decimal=4` the error tolerance. 

At higher sample sizes (`NSTEP=30000`) all the FFT based MSD tests fail while the standard "naive" ones do not. 

I would be super super grateful if you could help me get to the bottom of this! I may be missing something simple.

Reply:
Thank you for the details. I checked the test case to have a better idea of the error.

With NSTEP=30000 (and thus maximum value of the msd of 3*(NSTEP-1)**2 approx 2.7e9) the relative precision for decimal=4 is about 1e-13. FFTs seem to be more accurate than this (lose about one decimal place, so relative precision from about 2e-16 to 2e-15), but this is only the FFT part and the code does FFT + reverse FFT, so there goes one extra decimal place. The msd code involves other summation and difference operations on the series and will lose more accuracy to roundoff. I don't have the specific values though. (already checking FFTs accuracy estimates is interesting, see https://github.com/numpy/numpy/pull/14178 for instance for the NumPy test cases).

The test NSTEP=30000 passes with decimal=3, so 1e-12 of relative accuracy.

Reply:
Thanks for investigating @pdebuyl!
Makes much more sense in the context of **relative** precision, something which I wasn't thinking about (but probably should have).
I assume we are happy with a margin of ~1e-12 and will go ahead and refactor.

Reply:
Good to know! I don't know of an error estimate for the correlation algorithm for the MSD, if you find one at some point I would be interested indeed.